app_dart/lib/src/service/flags/dynamic_config.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport '../config.dart';
+/// @docImport 'dynamic_config_updater.dart';
+library;
+
+import 'dart:io' as io;
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+
+import 'content_aware_hashing_flags.dart';
+import 'dynamic_config_updater.dart';
+
+part 'dynamic_config.g.dart';
+
+/// Flags for the service that can be updated dynamically with out a restart.
+///
+/// Normally, this object is from [Config.flags], and is updated in the
+/// background automatically by [DynamicConfigUpdater], where the latest copy
+/// of `//app_dart/config.yaml` is fetched periodically.
+///
+/// To get a one-time copy of the file, see also:
+///
+/// - [DynamicConfig.fromRemoteLatest]
+/// - [DynamicConfig.fromLocalRepo]
+@JsonSerializable(explicitToJson: true)
+@immutable
+final class DynamicConfig {
+  /// Default configuration for flags.
+  static const defaultInstance = DynamicConfig._(
+    backfillerCommitLimit: 50,
+    contentAwareHashing: ContentAwareHashing.defaultInstance,
+  );
+
+  /// Upper limit of commit rows to be backfilled in API call.
+  ///
+  /// This limits the number of commits to be checked to backfill. When bots
+  /// are idle, we hope to scan as many commit rows as possible.
+  @JsonKey()
+  final int backfillerCommitLimit;
+
+  /// Flags associated with content-aware hashing.
+  @JsonKey()
+  final ContentAwareHashing contentAwareHashing;
+
+  const DynamicConfig._({
+    required this.backfillerCommitLimit,
+    required this.contentAwareHashing,
+  });
+
+  /// Creates [DynamicConfig] flags from a [json] object.
+  ///
+  /// Any omitted fields default to the values in [defaultInstance].
+  factory DynamicConfig({
+    int? backfillerCommitLimit,
+    ContentAwareHashing? contentAwareHashing,
+  }) {
+    return DynamicConfig._(
+      backfillerCommitLimit:
+          backfillerCommitLimit ?? defaultInstance.backfillerCommitLimit,
+      contentAwareHashing:
+          contentAwareHashing ?? defaultInstance.contentAwareHashing,
+    );
+  }
+
+  /// Creates [DynamicConfig] flags from a [json] object.
+  ///
+  /// Any omitted fields default to the values in [defaultInstance].
+  factory DynamicConfig.fromJson(Map<String, Object?>? json) {
+    return _$DynamicConfigFromJson(json ?? {});
+  }
+
+  /// Creates [DynamicConfig] flags from a [yaml] object.
+  factory DynamicConfig.fromYaml(YamlMap? yaml) {
+    return DynamicConfig.fromJson(yaml?.asMap);
+  }
+
+  /// Returns the latest copy of [DynamicConfig] fetched from tip-of-tree.
+  ///
+  /// Equivalent to a single call to [DynamicConfigUpdater.fetchDynamicConfig].
+  static Future<DynamicConfig> fromRemoteLatest() async {
```

Comment:
nit: don't need `async` if you don't await on anything.

Reply:
Done!